src/v/kafka/client/consumer.cc

```diff
@@ -341,10 +341,7 @@ consumer::offset_commit(std::vector<offset_commit_request_topic> topics) {
     } else { // set epoch for requests tps
         for (auto& t : topics) {
             for (auto& p : t.partitions) {
-                auto tp = model::topic_partition{t.name, p.partition_index};
-                auto leader = co_await _topic_cache.leader(tp);
-                auto broker = co_await _brokers.find(leader);
-                p.committed_leader_epoch = _fetch_sessions[broker].epoch();
+                p.committed_leader_epoch = kafka::invalid_leader_epoch;
```

Comment:
I kinda have two notes here:
- I thought this was in redpanda at first, but since this is in the client it should be fine to keep. The client knows of the leader epoch only through what is returned from a metadata response. So, if redpanda returns -1 properly, _using_ that -1 here should be fine (this does not explicitly need to set -1).
- Skimming the fetch_session.cc source, I do not think that fetch sessions are using the epoch correctly. Down below in `make_offset_commit_request`, as well as here, `_epoch` is a client-tracked epoch that is bumped every fetch response. That's not what the epoch is; it is not incremented client side.

So, this _should_ be -1 right now because redpanda (per this pr) will be returning -1 for the leader epoch. However a further issue is that this group consumer needs to use the broker-returned epoch, not generate its own.

Reply:
@twmb should this second issue be fixed in this PR? if no, can you and @BenPope create a ticket document what needs to be fixed in the client?